<h2>Why is this an issue?</h2>
<p>This rule detects cases where <code>Foos.TFoo</code> has been extended.</p>
<h3>Noncompliant Code Example</h3>
<pre>
TO DO
</pre>
<h3>Compliant Solution</h3>
<pre>
TO DO
</pre>
