<h2>Why is this an issue?</h2>
<p>This rule allows banning a string in method names. It will flag any method name that contains the
  forbidden string.</p>
<h3>Noncompliant Code Example</h3>
<pre>
TO DO
</pre>
<h3>Compliant Solution</h3>
<pre>
TO DO
</pre>
