<h2>Why is this an issue?</h2>
<p>
  When calling or declaring a method, an empty argument or parameter list is superfluous and can be
  omitted.
</p>
<h3>Exceptions</h3>
<p>
  This rule does not apply to procedural types or array constructors.
</p>
<h2>How to fix it</h2>
<p>Remove the superfluous parentheses:</p>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
FunctionWithoutArguments(); // Noncompliant
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
FunctionWithoutArguments;
</pre>
<h2>Resources</h2>
<ul>
  <li>
    <a href="https://docwiki.embarcadero.com/RADStudio/en/Delphi_Statements#Functions_and_procedures">
      Delphi's Object Pascal Style Guide: Statements
    </a>
  </li>
</ul>