<h2>Why is this an issue?</h2>
<p>
  A class without a parent implicitly inherits from TObject. In these cases, it is recommended to
  explicitly inherit from TObject for readability.
</p>
<h2>How to fix it</h2>
<p>Specify <code>TObject</code> as the class's ancestor:</p>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
type
  TMyClass = class // Noncompliant
    // ...
  end;
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
type
  TMyClass = class(TObject)
    // ...
  end;
</pre>
<h2>Resources</h2>
<ul>
  <li>
    <a href="http://docwiki.embarcadero.com/RADStudio/en/Classes_and_Objects_(Delphi)#TObject_and_TClass">
      RAD Studio documentation: TObject and TClass
    </a>
  </li>
</ul>
</p>
