<h2>Why is this an issue?</h2>
<p>
  For clarity and consistency, visibility sections within the body of a class, interface, or
  record declaration should be organized in the following order:
</p>
<ol>
  <li>Field declarations</li>
  <li>Method declarations</li>
  <li>Property declarations</li>
</ol>
<p>
  The Delphi compiler also forbids field declarations from appearing after method or property
  declarations in the same visibility section.
</p>
<h2>How to fix it</h2>
<p>Reorder the offending visibility section into fields, then methods, then properties:</p>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
type
  TMyType = class(TObject)
  private
    FMyVar: Integer;
    property MyVar: Integer read FMyVar;
    procedure DoThing; // Noncompliant
    procedure DoAnotherThing;
    property MyOtherVar: Integer read FMyVar;
  end;
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
type
  TMyType = class(TObject)
  private
    FMyVar: Integer;
    procedure DoThing;
    procedure DoAnotherThing;
    property MyVar: Integer read FMyVar;
    property MyOtherVar: Integer read FMyVar;
  end;
</pre>
<h2>Resources</h2>
<ul>
  <li>
    <a href="https://docwiki.embarcadero.com/RADStudio/en/Type_Declarations#Class_Body_Organization">
      Delphi's Object Pascal Style Guide: Type Declarations
    </a>
  </li>
</ul>
