<h2>Why is this an issue?</h2>
<p>
  Deeply nested subroutines make for code that is difficult to navigate and understand.
  While subroutines can be helpful for small units of work specific to a particular method, their
  ambiguous syntax and non-linear control flow make them confusing when containing large amounts of
  functionality.
</p>
<h2>How to fix it</h2>
<p>Extract the nested subroutines out into one or more independent methods:</p>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
procedure DoThing;
  procedure DoSubThing;
    procedure DoSubSubThing; // Noncompliant
    begin
      // ...
    end;
  begin
    // ...
  end;
begin
  // ...
end;
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
procedure DoSubThing;
  procedure DoSubSubThing;
  begin
    // ...
  end;
begin
  // ...
end;

procedure DoThing;
begin
  // ...
end;
</pre>