<h2>Why is this an issue?</h2>
<p>
  Terminating all Delphi statements with a semicolon allows them to be easily read, swapped around,
  and added to.
</p>
<p>
  While semicolons can be omitted for some statements immediately preceding an <code>end</code>,
  this is not recommended.
</p>
<h2>How to fix it</h2>
<p>Add the missing semicolon:</p>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
procedure Example;
begin
  Foo;
  Bar;
  Baz // Noncompliant
end;
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
procedure Example;
begin
  Foo;
  Bar;
  Baz;
end;
</pre>
<h2>Resources</h2>
<ul>
  <li>
    <a href="https://docwiki.embarcadero.com/RADStudio/en/Delphi_Statements">
      Delphi's Object Pascal Style Guide: Delphi Statements
    </a>
  </li>
</ul>