<h2>Why is this an issue?</h2>
<p>
  While Delphi is case-insensitive, casing of name declarations and references should be kept
  consistent to aid readability.
</p>
<h2>How to fix it</h2>
<p>Rename the identifier to match the casing in the original declaration:</p>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
procedure FooBar;
begin
  // ...
end;

procedure Example;
begin
  Foobar; // Noncompliant
end;
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
procedure FooBar;
begin
  // ...
end;

procedure Example;
begin
  FooBar;
end;
</pre>