<h2>Why is this an issue?</h2>
<p>
  <code>System.Assigned</code> is more flexible and has less edge cases than a normal nil
  comparison. In particular:
</p>
<ul>
  <li><code>Assigned</code> can be used to test if object procedures are assigned</li>
  <li>
    <code>Assigned</code> always checks the assignment of function references themselves -
    nil comparisons may call the function and check the assignment of its return value
  </li>
  <li><code>Assigned</code> is clearer and more self-documenting</li>
</ul>
<h2>How to fix it</h2>
<p>Replace the nil comparison with a call to <code>Assigned</code>:</p>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
procedure LogAssigned(MyObj: TObject);
begin
  if MyObj &lt;> nil then begin // Noncompliant
    Writeln('Object is assigned');
  end;
end;
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
procedure LogAssigned(MyObj: TObject);
begin
  if Assigned(MyObj) then begin
    Writeln('Object is assigned');
  end;
end;
</pre>
<h2>Resources</h2>
<ul>
  <li>
    <a href="http://docwiki.embarcadero.com/Libraries/en/System.Assigned">
      RAD Studio documentation: System.Assigned
    </a>
  </li>
</ul>