<h2>Why is this an issue?</h2>
<p>
  The Delphi project file (.dpr) contains the initialization logic for the program. It is
  partially generated by the IDE, is generally abstracted over by the .dproj file, and is
  underemphasized in editor tooling.
</p>
<p>
  Use of the project file for complex logic bloats the application entry point, makes the project
  more difficult to navigate, and reduces visibility.
</p>
<h2>How to fix it</h2>
<p>
  Move any complex logic into a method in a separate unit, which you can then reference from the
  initialization section of the project file:
</p>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
// MyProject.dpr
// ...
var
  MyVar: string; // Noncompliant
begin
  MyVar := Readln;
  Writeln(MyVar);
end.
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
// MyProject.dpr
// ...
uses MyUnit;

begin
  Foo;
end.

// MyUnit.pas
interface

procedure Foo;

implementation

procedure Foo;
var
  MyVar: string;
begin
  MyVar := Readln;
  Writeln(MyVar);
end;
</pre>