<h2>Why is this an issue?</h2>
<p>
  Parentheses should be used to clarify the intent behind a piece of code or enforce a desired
  order of operations. Redundant pairs of parentheses do neither of these things, making code more
  confusing and less readable.
</p>
<h2>How to fix it</h2>
<p>Remove the redundant parentheses:</p>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
Result := ((X + 1)); // Noncompliant
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
Result := (X + 1);
</pre>