<h2>Why is this an issue?</h2>
<p>
  Re-raising a caught exception explicitly (i.e. passing the exception as an argument) causes a
  double free, as Delphi does not know that the re-raised exception is the original one and
  frees it at the end of the exception handler.
</p>
<p>
  In these situations, a bare <code>raise</code> should be used. This correctly indicates that the
  current exception should be re-raised.
</p>
<h2>How to fix it</h2>
<p>Replace the <code>raise X</code> statement with a simple <code>raise</code>:</p>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
try
  DoSomething;
except
  on E: Exception do begin
    Log.Debug('Exception message = ' + E.Message);
    raise E; // Noncompliant
  end;
end;
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
try
  DoSomething;
except
  on E: Exception do begin
    Log.Debug('Exception message = ' + E.Message);
    raise;
  end;
end;
</pre>
<h2>Resources</h2>
<ul>
  <li>
    <a href="https://marc.durdin.net/2012/10/how-not-to-re-raise-an-exception-in-delphi/">
      Marc Durdin's Blog: How not to re-raise an exception in Delphi
    </a>
  </li>
  <li>
    <a href="http://delphi.org/2017/06/really-bad-exception-abuse/">
      Delphi.org: Exceptionally Bad Exception Abuse
    </a>
  </li>
</ul>