<h2>Why is this an issue?</h2>
<p>
  While past Delphi versions allowed a trailing comma in argument lists, more recent versions treat
  trailing commas as a parsing error and do not compile.
</p>
<p>
  Having trailing commas in argument lists makes code non-portable between different versions of
  Delphi and may pose a problem during upgrades.
</p>
<h2>How to fix it</h2>
<p>Remove the extraneous trailing comma:</p>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
Foo(1, 2, 3,); // Noncompliant
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
Foo(1, 2, 3);
</pre>
