<h2>Why is this an issue?</h2>
<p>
  Functions may omit the return type in their implementation if the return type has been defined
  in their declaration. This is confusing and unintuitive. For the sake of code readability,
  function return types should always be specified.
</p>
<h2>How to fix it</h2>
<p>
  Ensure that the function's implementation signature matches its declaration:
</p>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
interface

type TFoo = class(TObject)
  function Bar(Baz: Byte): Integer;
end;

implementation

function TFoo.Bar(Baz: Byte); // Noncompliant
begin
  // ...
end;
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
interface

type TFoo = class(TObject)
  function Bar(Baz: Byte): Integer;
end;

implementation

function TFoo.Bar(Baz: Byte): Integer;
begin
  // ...
end;
</pre>